cadence/contracts/AxelarGateway.cdc

```diff
@@ -0,0 +1,312 @@
+        return nil
+      }
+
+      return AxelarAuthWeighted.TransferOperatorshipParams(
+        newOperators : self.rawParams[0] as! [String],
+        newWeights : self.rawParams[1] as! [UInt256],
+        newThreshold : self.rawParams[2] as! UInt256,
+      )
+    }
+  }
+
+  access(all) resource interface Executable {
+    access(all) fun executeApp(commandResource: &AxelarGateway.CGPCommand, sourceChain: String, sourceAddress: String, payload: [UInt8])
+  }
+
+  access(all) fun callContract(sender: Address, destinationChain: String, destinationContractAddress: String, payload: [UInt8]) {
+    emit ContractCall(
+      sender: sender,
+      destinationChain: destinationChain,
+      destinationContractAddress: destinationContractAddress,
+      payloadHash: String.encodeHex(Crypto.hash(payload, algorithm: HashAlgorithm.KECCAK_256)),
+      payload: payload
+    )
+  }
+
+  access(all) fun isCommandExecuted(commandId: String): Bool {
+    return self.executedCommandIds.contains(commandId)
+  }
+
+  access(all) fun execute(
+    commandIds: [String],
+    commands: [String],
+    params: [[AnyStruct]],
+    operators: [String],
+    weights: [UInt256],
+    threshold: UInt256,
+    signatures: [String]
+  ) {
+    let encodedMessage = self.convertDataToHexEncodedMessage(commandIds: commandIds, commands: commands, params: params)
+
+    var allowOperatorshipTransfer = AxelarAuthWeighted.validateProof(message: encodedMessage, operators: operators, weights: weights, threshold: threshold, signatures: signatures)
+
+    let commandsLength = commandIds.length
+      
+    if (commandsLength != commands.length || commandsLength != params.length) {
+      panic("Invalid Commands")
+    }
+
+    var i = 0
+
+    while i < commandsLength {
+      var commandId = commandIds[i]
+      if (self.executedCommandIds.contains(commandId)) {
+        continue
+      }
+
+      let commandHash = Crypto.hash(commands[i].utf8, algorithm: HashAlgorithm.KECCAK_256)
+      let paramHandler = ParamHandler(params: params[i])
+
+      // Transfer OperatorShip is the only command that is directly processed by this contract - unlike other commannds
+      // it is used for managing gateway operators for the Axelar CGP protocol
+      if (commandHash == self.SELECTOR_TRANSFER_OPERATORSHIP && params[i].length == 3) {
+
+        if (allowOperatorshipTransfer) {
+          var transferOperatorShipParams = paramHandler.generateTransferOperatorshipParams()
+
+          if transferOperatorShipParams != nil {
+
+            // we only need to track this was executed, since there's no other processing for this command
+            self.executedCommandIds.append(commandId)
+            AxelarAuthWeighted.transferOperatorship(params: transferOperatorShipParams!)
+            emit Executed(commandId: commandId)
+          }
+        }
+      } else if (commandHash == self.SELECTOR_APPROVE_CONTRACT_CALL && params[i].length == 6) {
```

Comment:
Does this scope need any sort of signature verification? It looks like anyone could spam this and hydrate the `approvedCommands` field so long as the params are of proper type and length.

Reply:
The signature verification is actually conducted on line 159 within the `AxelarAuthWeighted.validateProof` method. This variable should've been a constant variable and we could possibly change the implementation to be a void method. This was how it was implemented on EVM side and we tried to keep it similar so that Axelar developers can have an easier time understanding the contract.